### PR TITLE
[MO] - [system test] -> try with resources, sals and tls authen kafka…

### PR DESCRIPTION
**Pull request by Seequick1**
### Type of change

- Enhancement / new feature
- Refactoring

### Description

This PR adding 
 - ST for testing `SASL with scram-sha mechanism with kafka connect` 
 - refactor creating kafka client inside of the try, `using try with resource feature`
 - exchange messages with `sasl security protocol` 
 - removing kafka-verifiable-producer.sh method, (bad approach - `creating producer inside kafka connect pod`)
### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [x] Write tests
- [x] Make sure all tests pass

### CODE REVIEW

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectST

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectST

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectST

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectST

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectST

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectST

**Comment by Seequick1** [comment]
All of these tests on local host passed. I assume, that could be some problem with resource.

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectST

**Comment by Seequick1** [comment]
@strimzi-ci run tests profile=regression testcase=ConnectST#testKafkaConnectWithFileSinkPlugin

File: systemtest/src/main/java/io/strimzi/systemtest/Constants.java [hunk 1]
```diff
@@ -13,7 +13,7 @@
     long TIMEOUT_FOR_DEPLOYMENT_CONFIG_READINESS = Duration.ofMinutes(7).toMillis();
     long TIMEOUT_FOR_RESOURCE_CREATION = Duration.ofMinutes(5).toMillis();
     long TIMEOUT_FOR_SECRET_CREATION = Duration.ofMinutes(2).toMillis();
-    long TIMEOUT_FOR_RESOURCE_READINESS = Duration.ofMinutes(12).toMillis();
+    long TIMEOUT_FOR_RESOURCE_READINESS = Duration.ofMinutes(14).toMillis();
```
Comment (ppatierno) [inline systemtest/src/main/java/io/strimzi/systemtest/Constants.java]: Out of curiosity why this change?

Comment (Seequick1) [reply systemtest/src/main/java/io/strimzi/systemtest/Constants.java]: I find out, that deployment of Kafka Connect sometimes needs approximately 13 minutes to deploy but on the other hand somethings it takes only 1 minute.

Comment (Frawless) [reply systemtest/src/main/java/io/strimzi/systemtest/Constants.java]: Unfortunately, it's a limitation of one from you QE environments and we can't do much with that. This is probably the easiest solution.

File: systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java [hunk 2]
```diff
@@ -822,7 +822,14 @@ public static void createSecretFromFile(String pathToOrigin, String key, String
         } catch (IOException e) {
             e.printStackTrace();
         }
+    }
 
-
+    public static void createFileSinkConnector(String poName, String topicName) {
```
Comment (Frawless) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java]:
```suggestion
    public static void createFileSinkConnector(String podName, String topicName) {
```

Comment (Frawless) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/StUtils.java]: Change it even in the lines below
